### PR TITLE
Dev work

### PR DESCRIPTION
**Pull request by v2xsami**

### CODE REVIEW

File: simulator/main.cpp [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+{
+public:
+    virtual void FillBuffer(v2x::SanePacketBuffer& buff) override
+    {
+        PID_POLLING_INFO obdData[v2x::PID_POLL_COUNT];
+
+        for (size_t i=0; i<v2x::PID_POLL_COUNT; ++i)
+        {
+            obdData[i].pid   = v2x::OBDParamTable[i].pid;
+            obdData[i].value = 0x123;
```
Comment (shumail92) [inline simulator/main.cpp]: I guess for now you are hardcoding `0x123` as value for all the obd params here? 
If so, good enough for now but then i guess we'll get rid of it once we replay the dump from txt file in future?

File: simulator/main.cpp [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+        }
+        GPS_DATA gps;
+        MEMS_DATA mems;
+
```
Comment (shumail92) [inline simulator/main.cpp]: Where are you filling up the `gps` & `mems` value?

File: simulator/main.cpp [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+        buff.Clear();
+        WriteFormatted(buff, "MSG=%s\n", event);
+        WriteFormatted(buff, "TS=%lu\n", mTimeAccumulator.count());
+        WriteFormatted(buff, "SK=%s\n", "foobar");
+        WriteFormatted(buff, "MAC=%s\n", mMac.c_str());
+        WriteFormatted(buff, "VIN=%s\n", mVin.c_str());
+        WriteFormatted(buff, "VER=%s\n", "1.0");
+        SendShit(buff);
+    }
+    void SendShit(const v2x::SanePacketBuffer& buff)
```
Comment (shumail92) [inline simulator/main.cpp]: 💯 for name `SendShit`

File: simulator/main.cpp [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+        }
+
+        for (auto& it : cars)
+        {
+            it->Quit();
+        }
+
+    }
+    catch (const std::exception& e)
+    {}
```
Comment (shumail92) [inline simulator/main.cpp]: probably we should print out the exception?

File: simulator/main.cpp (hunk 1 above)
Comment (v2xsami) [reply simulator/main.cpp]: Yeah, this is just filling some data there, semantically complete junk. 

Yeah, in future we could implement a new data generator that reads for example a database  dump file provides sensible data.

File: simulator/main.cpp (hunk 2 above)
Comment (v2xsami) [reply simulator/main.cpp]: I'm not. Implicitly filled by whateever garbage is in the stack.

File: simulator/main.cpp (hunk 4 above)
Comment (v2xsami) [reply simulator/main.cpp]: Yeah I'll add that.

Comment (v2xsami) [reply simulator/main.cpp]: @shumail92 done.

**Pull request merged**